### DISCUSSION TITLE
Github workflow branch patterns not working. Need help.

### DISCUSSION THREAD

**Discussion by tejaswini-imply** [community/community#52493] [Actions] [answered]
### Select Topic Area

Question

### Body

I want CI branches to trigger on the master and any release branches. Our release branches are in the format X.X.X or X.X.X-LTS(hyphen and one or more non-white-space chars), and X indicates any number of digits. e.g. `26.0.0` or `26.0.0-STS`

I was trying this snippet following the [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) 
```
branches:
      - master
      - '[0-9]+.[0-9]+.[0-9]+(-[^ ]+)?' # release branches
```
But the CI was failing with parsing error
```
push event contained invalid branches patterns: the following globs were invalid: [0-9]+.[0-9]+.[0-9]+(-[^ ]+)?, pull_request event contained invalid branches patterns: the following globs were invalid: [0-9]+.[0-9]+.[0-9]+(-[^ ]+)?
```

Can someone point out what I was doing different?

**Comment by airtower-luna** [comment, marked as answer]
It looks to me like this part at the end is the problem: `(-[^ ]+)?` The filter pattern is just an extended glob, not a regular expression with groups. The closest I can think of right now is to use two patterns, e.g. like so (you might want to adjust the second one if you want to allow characters other than alphanumeric ones):

```yaml
      - '[0-9]+.[0-9]+.[0-9]+'
      - '[0-9]+.[0-9]+.[0-9]+-[A-Za-z0-9]+'
```

**Reply by tejaswini-imply** [reply]
Thanks for the response. This should do for now.
